qiskit/ignis/experiments/calibration/instruction_data/gate_and_schedules_definition.py

```diff
@@ -0,0 +1,140 @@
+            qubits: Set of qubits to which the instruction applies.
+
+        Returns:
+            A QuantumCircuit with parameters in it.
+        """
+        return self._instructions[(name, qubits)]
+
+    def get_instruction(self, name: str, qubits: Iterable,
+                        free_parameters: List = None) -> QuantumCircuit:
+        """
+        Returns the QuantumCircuit of the instruction where all parameters aside for those
+        explicitly specified are bound to their calibration.
+
+        Args:
+            name: name of the instruction to retrive
+            qubits: qubits to which the instruction applies.
+            free_parameters: Parameters that should be left unbound. If None is specified then
+                all parameters will be bound to their calibrated values.
+        """
+        # TODO Get the parameters
+
+        # TODO Get their values
+
+        # TODO return circ with parameters assigned
+        raise NotImplemented
+
+    def create_basic_instruction(self, name: str, duration: int, pulse_envelope: ParametricPulse,
+                                 channel: PulseChannel, params = None):
+        """
+        Creates a basic instruction in the dictionary.
+
+        Args:
+            name:
+            duration: The duration of the pulse schedule.
+            pulse_envelope: The callable pulse envelope. Its parameters are given by
+                **params after duration is added to it.
+            channel: The channel to which we apply the basic instruction.
+        """
+        # Avoids having the user manage the parameters.
+        if not params:
+            params = self._basic_inst_parameters(name, channel, pulse_envelope)
+
+        gate = Gate(name=name, num_qubits=1, params=[_ for _ in params.values()])
+
+        if duration:
+            params['duration'] = duration
+
+        schedule = Schedule(name=name)
+        schedule = schedule.insert(0, Play(pulse_envelope(**params), channel))
+
+        if isinstance(channel, ControlChannel):
+            # TODO Need a mapping from ControlChannel to qubits
+            # TODO This works for CR, would it work for TC?
+            raise NotImplemented
+        else:
+            qubit = int(channel.name.replace('d', '').replace('m', ''))
+
+        circ = QuantumCircuit(1)
+        circ.append(gate, [0])
+        circ.add_calibration(gate, [qubit], schedule)
+
+        self._instructions[(name, (qubit))] = circ
+
+    def _basic_inst_parameters(self, name: str, channel: PulseChannel,
+                               pulse_envelope: ParametricPulse) -> Dict[str, ParameterExpression]:
+        """
+        This functions avoids the user to have to manage parameters and their name.
+
+        TODO Is there a better way of doing this?
+        """
+        params = {}
+        if pulse_envelope.__name__ in ['Drag', 'Gaussian', 'GaussianSquare']:
+            params['amp'] = Parameter(name + '.' + channel.name + '.amp')
```

Comment:
seems like this is defined for physical channel. Can we reduce namespace and make it logical channel? i.e. we can reduce (almost) duplicated entries in the dictionary.

Reply:
Can you make a suggestion? My intention here is to be able to do e.g. `Drag(**params)` and to have parameter names that we can use to easily refer to in the pulse parameter table.

Reply:
how about this?
```
define Pulse Instance rather than Instruction
   X90p: Gaussian(duration, amp, sigma)

a method that create Instruction on the fly
   create('x90p', qubits=0) -> Play(X90p_this, DriveChannel(channel_map(qubit=0)))
   ---
   X90p_this = X90p.assign_parameter({amp: amp.q0.d0.gaussian})
```

user can avoid inserting almost the same instruction for multiple channels.

Reply:
What if you want `x90p` to be a Drag pulse on qubit 0 and a Gaussian pulse on qubit 1?